### PR TITLE
Enable the developed CI, added Jupyter notebook for temp-sense gen (WIP) and dependencies.sh script

### PR DESCRIPTION
**Pull request by saicharan0112**
* Enabled the developed CI. Was waiting for the OpenROAD scripts to get updated which happened in PR https://github.com/idea-fasoc/OpenFASOC/pull/82.
* Updated docs and added Jupyter notebook for temp-sense. Though WIP because of the missing klayout to merge gds, the foundation is build I guess.
* Updated Dependencies.sh script to be check the python version and be flexible with the python package installations, also build ngspice from the source. (Addressed - https://github.com/idea-fasoc/OpenFASOC/issues/85)

Following this PR, below are the to-do items, in the same order if possible:
* Improve Jupyter notebooks by bringing klayout and simulations into the environment
* Address https://github.com/idea-fasoc/OpenFASOC/pull/88#issuecomment-1227422210 and https://github.com/idea-fasoc/OpenFASOC/issues/78
* Address https://github.com/idea-fasoc/OpenFASOC/issues/1

### CODE REVIEW

**Comment by saicharan0112** [comment]
@msaligane I want to update the jupyter notebook. Please don't merge this PR yet. In the meanwhile, please feel free to edit the docs especially "Things to Improve" section https://github.com/saicharan0112/OpenFASOC/blob/main/docs/source/developers-guide.rst

**Comment by saicharan0112** [comment]
The Jupyter notebook is working fine. klayout is the only missing piece to complete the generation of the merged gds file along with lvs and drc checks.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,32 +6,33 @@ on:
...
   workflow_dispatch:
-  # schedule:
-  # - cron: 0 1 * * * # run at 1 AM UTC
+  schedule:
+   - cron: 0 1 * * * # run at 1 AM UTC
 
 jobs:
   build_docker_image:
-    runs-on: self-hosted
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (saicharan0112) [reply .github/workflows/ci.yml:14]: Yeah, but it doesn't matter cuz the testing is done inside the docker container.

**Comment by msaligane** [comment]
Can you fix the pre-commit. I think I screwed it up. I will merge then.
Great job @saicharan0112 !

**Comment by saicharan0112** [comment]
> Can you fix the pre-commit. I think I screwed it up. I will merge then. Great job @saicharan0112 !

Done. Also, FYI, the commits done with the message "
Updated README with latest stable version numbers of dependent tools" is by the GHA bot. It is testing the temp-sense flow with the latest tool versions and updating the conda_versions.txt, versions.txt and the README file. 
This is scheduled to happen at 1 AM PST everyday

**Review by msaligane — APPROVED** [review]

**Comment by saicharan0112** [comment]
Thanks for merging the PR!

**Pull request merged**